deepmd/utils/transfer_to_ascend.py

```diff
@@ -0,0 +1,147 @@
+        if speccified log will be written to this file
+    log_level : int
+        logging level
+    """
+    try:
+        t_jdata = get_tensor_by_name(input, 'train_attr/training_script')
+        t_min_nbor_dist = get_tensor_by_name(input, 'train_attr/min_nbor_dist')
+        jdata = json.loads(t_jdata)
+    except GraphWithoutTensorError as e:
+        if training_script == None:
+            raise RuntimeError(
+                "The input transfering model: %s has no training script or min_nbor_dist information, "
+                "which is not supported by the transfer_to_ascend interface. "
+                "Please consider using the --training-script command within the transfer_to_ascend interface to provide the training script of the input transfering model. "
+                "Note that the input transfering script must contain the correct path to the training data." % input
+            ) from e
+        elif not os.path.exists(training_script):
+            raise RuntimeError(
+                "The input training script %s (%s) does not exist! Please check the path of the training script. " % (input, os.path.abspath(input))
+            ) from e
+        else:
+            log.info("stage 0: compute the min_nbor_dist")
+            jdata = j_loader(training_script)
+            jdata = update_deepmd_input(jdata)
+            t_min_nbor_dist = get_min_nbor_dist(jdata, get_rcut(jdata))
+
+    _check_transfer_model_type(input)
+
+    tf.constant(t_min_nbor_dist,
+        name = 'train_attr/min_nbor_dist',
+        dtype = GLOBAL_ENER_FLOAT_PRECISION)
+    jdata["model"]["descriptor"]["precision"] = "float16"
+    jdata["model"]["fitting_net"]["precision"] = "float16"
+    jdata["model"]["transfered_from_model"] = True
+    jdata["training"]["save_ckpt"] = os.path.join("model-transfer", "model.ckpt")
+    jdata = update_deepmd_input(jdata)
+    jdata = normalize(jdata)
+
+    # check the descriptor info of the input file
+    # move to the specific Descriptor class
+
+    # stage 1: training or refining the model with tabulation
+    log.info("\n\n")
+    log.info("stage 1: generate the mix-precision model")
+    control_file = "ascend-transfer.json"
+    with open(control_file, "w") as fp:
+        json.dump(jdata, fp, indent=4)
+    try:
+        train(
+            INPUT=control_file,
+            init_model=None,
+            restart=None,
+            init_frz_model=input,
+            output=control_file,
+            mpi_log=mpi_log,
+            log_level=log_level,
+            log_path=log_path,
+            is_compress=False,
+            skip_neighbor_stat=True,
+        )
+    except GraphTooLargeError as e:
+        raise RuntimeError(
+            "The uniform step size of the tabulation's first table is %f, " 
+            "which is too small. This leads to a very large graph size, "
+            "exceeding protobuf's limitation (2 GB). You should try to "
+            "increase the step size." % step
```

Comment:
`step` is not defined.

Reply:
Thanks for the comment, we have modified the error explanation.